development/extensions/tutorial_notifications.rst

```diff
@@ -0,0 +1,1190 @@
+=======================
+Tutorial: Notifications
+=======================
+
+.. contents:: This tutorial explains:
+   :depth: 1
+   :local:
+
+Introduction
+============
+The notification system was introduced with the release of phpBB :guilabel:`3.1`.
+Since then it has seen further developments and enhancements for users and extension developers.
+In :guilabel:`3.2` the notification system was heavily refactored making it faster and more efficient.
+
+phpBB users have complete control over their personal notification interactions.
+They can set their own preferences for notification methods or disable them all together.
+Mark notifications as read when necessary, even though in a lot of cases this is done automatically.
+
+In phpBB there are two default notification methods: email-based and board-based notifications.
+The email method is only available when the ''board-wide emails'' setting has been enabled by the board administrator in the ACP.
+Extensions, however, can create and add their own notification methods.
+
+This tutorial will document how and extension can leverage the notification system by creating a notification
+based on some sort of triggering event (such as replying to a post),
+sending the notification via either the board, email or some other custom methods, and managing its
+notifications.
+
+File Structure
+--------------
+This is an overview of the file structure needed for a notification to function within an extension.
+In total there are 7 files that will be covered in this tutorial, to create and send notifications.
+These files will be referenced throughout this tutorial.
+
+
+| |fa-user| vendor
+| └─ |fa-ext| extension
+|  |nbsp| ├─ |fa-folder| config
+|  |nbsp| │ |nbsp| └─ |fa-file| services.yml
+|  |nbsp| ├─ |fa-folder| event
+|  |nbsp| │ |nbsp| └─ |fa-file| listener.php
+|  |nbsp| ├─ |fa-folder| language
+|  |nbsp| │ |nbsp| |nbsp| └─ |fa-folder| en
+|  |nbsp| │ |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| └─ |fa-folder| email
+|  |nbsp| │ |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| │ |nbsp| |nbsp| └─ |fa-folder| short
+|  |nbsp| │ |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| │ |nbsp| |nbsp| │ |nbsp| |nbsp| └─ |fa-file| sample.txt
+|  |nbsp| │ |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| │ |nbsp| |nbsp| └─ |fa-file| sample.txt
+|  |nbsp| │ |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| └─ |fa-file| extension_common.php
+|  |nbsp| ├─ |fa-folder| notification
+|  |nbsp| │ |nbsp| |nbsp| └─ |fa-folder| type
+|  |nbsp| │ |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| |nbsp| └─ |fa-file| sample.php
+|  |nbsp| └─ |fa-file| ext.php
+|
+
+Creating a Notification Type
+============================
+The first thing is to determine a name for the type of notification we will create.
+For the purposes of this tutorial we will name it ''sample''. All extensions should always
+use their vendor and name as prefixes, so the notification type name we'll be using is: ``vendor.extension.notification.type.sample``.
```

Comment:
Maybe use acme.demo as name like in https://area51.phpbb.com/docs/dev/3.1.x/extensions/tutorial_key_concepts.html

Reply:
As far as I am aware, we are trying to move away from `acme/demo` to `vendor/extension`. So it's truly clear what is being meant, cause while most of us might know that acme is used as a 'fictional extension writer', it isn't really clear for everyone. While it should be 'obvious to anyone  looking at it, even for the first time'.
Perhaps @VSEphpbb can way in.

Reply:
I don't like using acme demo.